src/dmd/frontend.h

```diff
@@ -1946,8 +1946,6 @@ typedef Array<TemplateInstance* > TemplateInstances;
}\x0a\x0a    override void visit (TypeInfoVectorDeclaration _)\x0a    {\x0a        typeName = \"TypeInfoVectorDeclaration\";\x0a    }\x0a\x0a    override void visit (FuncAliasDeclaration _)\x0a    {\x0a        typeName = \"FuncAliasDeclaration\";\x0a    }\x0a\x0a    override void visit (ErrorInitializer _)\x0a    {\x0a        typeName = \"ErrorInitializer\";\x0a    }\x0a\x0a    override void visit (ErrorExp _)\x0a    {\x0a        typeName = \"ErrorExp\";\x0a    }\x0a\x0a    override void visit (ComplexExp _)\x0a    {\x0a        typeName = \"ComplexExp\";\x0a    }\x0a\x0a    override void visit (StructLiteralExp _)\x0a    {\x0a        typeName = \"StructLiteralExp\";\x0a    }\x0a\x0a    override void visit (ObjcClassReferenceExp _)\x0a    {\x0a        typeName = \"ObjcClassReferenceExp\";\x0a    }\x0a\x0a    override void visit (SymOffExp _)\x0a    {\x0a        typeName = \"SymOffExp\";\x0a    }\x0a\x0a    override void visit (OverExp _)\x0a    {\x0a        typeName = \"OverExp\";\x0a    }\x0a\x0a    override void visit (HaltExp _)\x0a    {\x0a        typeName = \"HaltExp\";\x0a    }\x0a\x0a    override void visit (DotTemplateExp _)\x0a    {\x0a        typeName = \"DotTemplateExp\";\x0a    }\x0a\x0a    override void visit (DotVarExp _)\x0a    {\x0a        typeName = \"DotVarExp\";\x0a    }\x0a\x0a    override void visit (DelegateExp _)\x0a    {\x0a        typeName = \"DelegateExp\";\x0a    }\x0a\x0a    override void visit (DotTypeExp _)\x0a    {\x0a        typeName = \"DotTypeExp\";\x0a    }\x0a\x0a    override void visit (VectorExp _)\x0a    {\x0a        typeName = \"VectorExp\";\x0a    }\x0a\x0a    override void visit (VectorArrayExp _)\x0a    {\x0a        typeName = \"VectorArrayExp\";\x0a    }\x0a\x0a    override void visit (SliceExp _)\x0a    {\x0a        typeName = \"SliceExp\";\x0a    }\x0a\x0a    override void visit (ArrayLengthExp _)\x0a    {\x0a        typeName = \"ArrayLengthExp\";\x0a    }\x0a\x0a    override void visit (DelegatePtrExp _)\x0a    {\x0a        typeName = \"DelegatePtrExp\";\x0a    }\x0a\x0a    override void visit (DelegateFuncptrExp _)\x0a    {\x0a        typeName = \"DelegateFuncptrExp\";\x0a    }\x0a\x0a    override void visit (DotExp _)\x0a    {\x0a        typeName = \"DotExp\";\x0a    }\x0a\x0a    override void visit (IndexExp _)\x0a    {\x0a        typeName = \"IndexExp\";\x0a    }\x0a\x0a    override void visit (ConstructExp _)\x0a    {\x0a        typeName = \"ConstructExp\";\x0a    }\x0a\x0a    override void visit (BlitExp _)\x0a    {\x0a        typeName = \"BlitExp\";\x0a    }\x0a\x0a    override void visit (RemoveExp _)\x0a    {\x0a        typeName = \"RemoveExp\";\x0a    }\x0a\x0a    override void visit (ClassReferenceExp _)\x0a    {\x0a        typeName = \"ClassReferenceExp\";\x0a    }\x0a\x0a    override void visit (VoidInitExp _)\x0a    {\x0a        typeName = \"VoidInitExp\";\x0a    }\x0a\x0a    override void visit (ThrownExceptionExp _)\x0a    {\x0a        typeName = \"ThrownExceptionExp\";\x0a    }\x0a}";
```

Comment:
If I need a rationale for why `extern(D)` enums should be ignored, this would be it.

Reply:
Except when it's used in another declaration ... (e.g. default values for arguments)

Reply:
> Except when it's used in another declaration ... (e.g. default values for arguments)

Can't we just ignore the symbol in that case and go straight to value?  Strings will already be stored statically, there's no need to add an extra symbol to point to it.

Reply:
Probably. Violating the DRY principle is not ideal for human readers but I think they are more likely to read the D source anyway.